src/utils/temp.ts

```diff
@@ -0,0 +1,35 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import path from 'path';
+
+function platformTempFolder(): string {
+	const tempBaseFolder = process.env['XDG_RUNTIME_DIR'] ?? os.homedir();
+	return os.platform() === 'win32'
+		? path.join(tempBaseFolder, 'ardrive-temp')
+		: path.join(tempBaseFolder, '.ardrive', 'temp');
+}
+/**
+ * Gets a folder path for storing temporary files.
+ */
+export function getTempFolder(): string {
+	const tempFolderPath = platformTempFolder();
+	if (fs.existsSync(tempFolderPath)) {
+		return tempFolderPath;
+	}
+
+	fs.mkdirSync(tempFolderPath, { recursive: true });
+
+	return tempFolderPath;
+}
+
+export function cleanUpTempFolder(): void {
+	const tempFolderPath = platformTempFolder();
+	fs.readdir(tempFolderPath, (err, files) => {
+		if (err) throw err;
+		for (const file of files) {
+			fs.unlink(path.join(tempFolderPath, file), (err) => {
+				if (err) throw err;
```

Comment:
I don't mind this syntax, but as a standard we've always been using the brackets for more readable executions:

```suggestion
		if (err) { 
		    throw err;
		}
		for (const file of files) {
			fs.unlink(path.join(tempFolderPath, file), (err) => {
				if (err) {
				    throw err;
				}    
```